src/qtt/algorithms/generic.py

```diff
@@ -93,13 +97,16 @@ def subpixelmax(A, mpos, verbose=0):
     cy = val
     ay = (valm + valp) / 2 - cy
     by = ay + cy - valm
-    shift = -by / (2 * ay)  # Maxima of quadradic
+
+    if np.any(ay == 0):
+        shift = 0 * ay
+    else:
+        shift = -by / (2 * ay)  # Maxima of quadradic
 
     if verbose:
         print('subpixelmax: mp %d, pp %d\n', mp, pp)
```

Comment:
The print statement is not correct:
```suggestion
        print('subpixelmax: mp %d, pp %d\n' % (mp, pp))
```